test/TransformSpec.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE QuasiQuotes #-}
+module TransformSpec where
+
+import Frost
+import Frost.Plugin
+import Frost.Plugins (transform)
+
+import Text.Pandoc
+import Data.Function ((&))
+import Polysemy
+import Polysemy.Error
+import qualified Data.Text as T
+import Test.Hspec
+import Text.RawString.QQ
+
+
+purgePlugin :: Plugin r
+purgePlugin = Plugin { pluginName = "null"
+                    , substitute = \_ -> return []
+                    , addToMeta = \m -> return nullMeta
+                    }
+
+textPlugin :: String -> Plugin r
+textPlugin text= Plugin { pluginName = "text:insert"
+                    , substitute = \_ -> return [Plain [Str text]]
+                    , addToMeta = \m -> return nullMeta
+                    }
+
+doublePlugin ::  Plugin r
+doublePlugin= Plugin { pluginName = "double"
+                    , substitute = \i -> return [Plain [Str $ show $ 2 * read i]]
+                    , addToMeta = \m -> return nullMeta
+                    }
+
+spec :: Spec
+spec =
+  describe "Frost.Plugins transform" $ do
+    it "should keep document as is, if no frost code blocks in it" $ do
+      -- given
+      let blocks = [ HorizontalRule
```

Comment:
Wouldn't it make more sense to generate arbitrary `[blocks]` or even whole `Pandoc` with meta?

Reply:
I thought about it. Especially since `pandoc-types` comes with whole packet of `Arbirary` instances for all pandoc types. 
My only concern was that in an infinitely long universe QuickSort would eventually generate a doc with a code block that matches the plugin, and the test would fail.